[Tests] Lambda related tests failed

<!-- Describe the bug report / feature request here -->

Found this problem in #3804. The failure seems not related to this PR but instead more related to #3801?

https://github.com/skypilot-org/skypilot/actions/runs/10260808607/job/28387454238

```python
=================================== FAILURES ===================================
______________________ test_instance_type_from_cpu_memory ______________________
[gw0] linux -- Python 3.8.18 /opt/hostedtoolcache/Python/3.8.18/x64/bin/python
tests/test_optimizer_dryruns.py:277: in test_instance_type_from_cpu_memory
    assert 'gpu_1x_rtx6000' in stdout  # Lambda, 14 vCPUs, 46 GB memory
E   AssertionError: assert 'gpu_1x_rtx6000' in 'D 08-06 05:33:38 optimizer.py:265] #### Task<name=test_task>(run=<empty>)\r\nD 08-06 05:33:38 optimizer.py:265]   res...proxy_command: None\nI 08-06 05:33:40 execution.py:283] Dryrun finished.\nClusters\nNo existing clusters.\n\n\x1b[?25h'
```


<!-- If relevant, fill in versioning info to help us troubleshoot -->

--- a/tests/test_optimizer_dryruns.py
+++ b/tests/test_optimizer_dryruns.py
@@ -274,7 +274,7 @@ def test_instance_type_from_cpu_memory(monkeypatch, capfd):
     assert 'n2-highcpu-4' in stdout  # GCP, 4 vCPUs, 4 GB memory
     assert 'c6i.xlarge' in stdout  # AWS, 4 vCPUs, 8 GB memory
     assert 'Standard_F4s_v2' in stdout  # Azure, 4 vCPUs, 8 GB memory
-    assert 'gpu_1x_rtx6000' in stdout  # Lambda, 14 vCPUs, 46 GB memory
+    assert 'cpu_4x_general' in stdout  # Lambda, 4 vCPUs, 16 GB memory
 
     _test_resources_launch(monkeypatch, accelerators='T4')
     stdout, _ = capfd.readouterr()

